eng/common/pipelines/templates/jobs/sdk-generation.yml

```diff
@@ -0,0 +1,160 @@
+parameters:
+  - name: SpecRepoBranch
+    type: string
+    default: 'main'
+  - name: SpecRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: SdkRepoOwner
+    type: string
+  - name: SdkRepoName
+    type: string
+
+  - name: SdkRepoNameForSpecGen
+    type: string
+  - name: SdkRepoBranch
+    type: string
+    default: 'main'
+  - name: ConfigType
+    type: string
+    values:
+      - 'TypeSpec'
+      - 'AutoRest'
+  - name: TspConfigPath
+    type: string
+  - name: ReadmePath
+    type: string
+  - name: ApiVersion
+    type: string
+    default: 'not-specified'
+  - name: CodePushRequired
+    type: boolean
+  - name: SdkPrRequired
+    type: boolean
+  - name: ApiViewRequired
+    type: boolean
+
+jobs:
+- job:
+  displayName: 'SDK Generation'
+  variables:
+    - template: /eng/pipelines/templates/variables/image.yml
+    - name: SpecRepoUrl
+      value: 'https://github.com/Azure/azure-rest-api-specs'
+    - name: SpecRepoFullName
+      value: 'Azure/azure-rest-api-specs'
+    - name: SpecRepoName
+      value: 'azure-rest-api-specs'
+    - name: SpecRepoDirectory
+      value: $(System.DefaultWorkingDirectory)/$(SpecRepoName)
+    - name: SdkRepoDirectory
+      value: $(System.DefaultWorkingDirectory)/${{ parameters.SdkRepoName }}
+  pool:
+    name: $(LINUXPOOL)
+    vmImage: $(LINUXVMIMAGE)
+
+  steps:
+    - checkout: none
+
+    - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+      parameters:
+        Paths:
+        - '/*'
+        - '!sdk/**/test-recordings/*'
+        - '!sdk/**/recordings/*'
+        - '!sdk/**/SessionRecords/*'
+        - '!sdk/**/session-records/*'
+        Repositories:
+        - Name: $(SpecRepoFullName)
+          Commitish: ${{ parameters.SpecRepoBranch }}
+          WorkingDirectory: $(SpecRepoDirectory)
+        - Name: ${{ parameters.SdkRepoOwner }}/${{ parameters.SdkRepoName }}
+          Commitish: ${{ parameters.SdkRepoBranch }}
+          WorkingDirectory: $(SdkRepoDirectory)
+
+    - script: |
+        if [ "${{ parameters.SpecRepoCommit }}" = "HEAD" ]; then
+          cd $(SpecRepoDirectory)
+          default_commit=$(git rev-parse HEAD)
+          echo "##vso[task.setvariable variable=SpecRepoCommit]$default_commit"
+          echo "SpecRepoCommit variable set to default commit: $default_commit"
+        else
+          echo "##vso[task.setvariable variable=SpecRepoCommit]${{ parameters.SpecRepoCommit }}"
+          echo "SpecRepoCommit variable set to: ${{ parameters.SpecRepoCommit }}"
+        fi
+      displayName: 'Set SpecRepoCommit variable'
+
+    - task: NodeTool@0
+      inputs:
+        versionSpec: '22.x'
+      displayName: 'Install Node.js'
+
```

Comment:
Could you add a step to install python? Pls refer to the [setupPipeline](https://devdiv.visualstudio.com/DevDiv/_git/openapi-alps?path=/private/azure-rest-api-specs-pipeline/.azure-pipelines/templates/SetupPipeline.yml&version=GBmain&line=24&lineEnd=25&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) for more info.